app/src/main/java/org/simple/clinic/patient/recent/RecentPatientsView.kt

```diff
@@ -33,16 +51,22 @@ class RecentPatientsView(context: Context, attrs: AttributeSet) : FrameLayout(co
       adapter = groupAdapter
     }
 
-    Observable.mergeArray(screenCreates())
+    Observable.mergeArray(screenCreates(), adapterEvents)
         .compose(controller)
         .takeUntil(RxView.detaches(this))
         .observeOn(AndroidSchedulers.mainThread())
         .subscribe { uiChange -> uiChange(this) }
+
+    adapterEvents
+        .ofType<RecentPatientItemClicked>()
+        .takeUntil(RxView.detaches(this))
+        .subscribe { openPatientSummary(it.patientUuid) }
   }
 
   private fun screenCreates() = Observable.just(ScreenCreated())
 
   fun updateRecentPatients(recentPatients: List<RecentPatientItem>) {
+    recentPatients.forEach { it.uiEvents = adapterEvents }
```

Comment:
this too should be part of the controller maybe?

Reply:
This cannot be a part of the controller as this a UI related logic. Also, the controller is agnostic to the `adapterEvents`.
Refer `PrescribedDrugScreen#populateDrugsList()` for this.